apis/tempo/v1alpha1/microservices_types.go

```diff
@@ -162,9 +162,34 @@ const (
 	Dynamic ModeType = "dynamic"
 )
 
+// MultitenancySpec defines if gateway, qurier and db multitenancy support should
+// be enabled and which header is used to identify tenants.
+type MultitenancySpec struct {
+	// Enable multi tenancy support in tempo. Default is false.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default:=false
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Multitenancy enabled"
+	Enabled bool `json:"enabeld"`
```

Comment:
It probably needs more thinking what `enabled` flags are in the CR for multitenancy. There is already `spec.template.gateway.enabled` , we should make sure that the config is easy and does not require flipping multiple enabled flags for a single use-case.

Reply:
Some form of validation/defaulting to make the configuration easier.